src/client/Microsoft.Identity.Client/MsalException.cs

```diff
@@ -170,19 +172,82 @@ public override string ToString()
 
         // DEPRECATE / OBSOLETE - this functionality is not used and should be removed in a next major version
 
-        private const string ExceptionTypeKey = "type";
-        private const string ErrorCodeKey = "error_code";
-        private const string ErrorDescriptionKey = "error_description";
+        private class ExceptionSerializationKey
+        {
+            internal const string ExceptionTypeKey = "type";
+            internal const string ErrorCodeKey = "error_code";
+            internal const string ErrorDescriptionKey = "error_description";
+            internal const string AdditionalExceptionData = "additional_exception_data";
+            internal const string BrokerErrorContext = "broker_error_context";
+            internal const string BrokerErrorTag = "broker_error_tag";
+            internal const string BrokerErrorStatus = "broker_error_status";
+            internal const string BrokerErrorCode = "broker_error_code";
+            internal const string BrokerTelemetry = "broker_telemetry";
+        }
 
-        internal virtual void PopulateJson(JObject jobj)
+        internal virtual void PopulateJson(JObject jObject)
         {
-            jobj[ExceptionTypeKey] = GetType().Name;
-            jobj[ErrorCodeKey] = ErrorCode;
-            jobj[ErrorDescriptionKey] = Message;
+            jObject[ExceptionSerializationKey.ExceptionTypeKey] = GetType().Name;
+            jObject[ExceptionSerializationKey.ErrorCodeKey] = ErrorCode;
+            jObject[ExceptionSerializationKey.ErrorDescriptionKey] = Message;
+
+            var exceptionData = new JObject();
```

Comment:
nit:For this method and maybe the one below it, I would put these if statements in a seperate method and call them something like `PopulateWithBrokerExceptionData`. Especially since we may end up adding more stuff here in the future.

Reply:
I added comments before these code blocks instead. Tried extracting into a method but method names become too long to read.